src/tests/BuildWasmApps/Wasm.Build.Tests/ICUShardingTests.cs

```diff
@@ -0,0 +1,93 @@
+                    new object?[] { enableSharding, "am-ET", "ja-JP", true},    //no_CJK
+                    new object?[] { enableSharding, "am-ET", "vi-VN", false},   //no_CJK
+                    new object?[] { enableSharding, null, "zh-Hans", true})     //ALL
+                .WithRunHosts(host)
+                .UnwrapItemsAsArrays();
+
+        [Theory]
+        [MemberData(nameof(ICUShardingTestData), parameters: new object[] { true, RunHost.All })]
+        public void ShardingTests(BuildArgs buildArgs, bool enableSharding, string defaultCulture, string testCulture, bool expectToThrow, RunHost host, string id)
+            => TestICUSharding(buildArgs, enableSharding, defaultCulture, testCulture, expectToThrow, host, id);
+        
+        void TestICUSharding(BuildArgs buildArgs,
+                             bool enableSharding,
+                             string defaultCulture,
+                             string testCulture,
+                             bool expectToThrow,
+                             RunHost host,
+                             string id,
+                             bool? dotnetWasmFromRuntimePack=null)
+        {
+            string projectName = $"sharding_{enableSharding}_{defaultCulture}_{testCulture}";
+            string programText = $@"
+                using System;
+                using System.Globalization;
+                using System.Threading.Tasks;
+
+                public class TestClass {{
+                    public static int Main()
+                    {{
+                        try {{
+                            var culture = new CultureInfo(""{testCulture}"", false);
+                            Console.WriteLine(culture.Name);
+                        }}
+                        catch {{
+                            Console.WriteLine(""Culture Not Found"");
+                        }}
+                        
+                        return 42;
+                    }}
+                }}";
+
+            buildArgs = buildArgs with { ProjectName = projectName, ProjectFileContents = programText };
+            buildArgs = GetBuildArgsWith(buildArgs, extraProperties: $"<EnableSharding>{enableSharding}</EnableSharding><ICUDefaultCulture>{defaultCulture}</ICUDefaultCulture>");
+            if (dotnetWasmFromRuntimePack == null)
+                dotnetWasmFromRuntimePack = !(buildArgs.AOT || buildArgs.Config == "Release");
+
+            BuildProject(buildArgs,
+                        initProject: () => File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), programText),
+                        id: id,
+                        dotnetWasmFromRuntimePack: dotnetWasmFromRuntimePack);
+
+            string expectedOutputString = expectToThrow == true || enableSharding == false
+                                            ? "Culture Not Found"
+                                            : $"{CultureInfo.GetCultureInfo(testCulture).Name}";
```

Comment:
What happens if `enableSharding==false`? Why does it also result in `Culture not found`?

Reply:
Ah wait it should be the other way around sorry.